File: native/mixdog-graph/tests/fixtures/parity/objc/sample.m
// Parity fixture: declaration and import shapes the graph reports for Obj-C.
#import <Foundation/Foundation.h>
#import "Store.h"
@import UIKit;

@protocol Storage <NSObject>
- (NSString *)read:(NSString *)key;
@end

@interface Store : NSObject <Storage>
@property(nonatomic, copy) NSString *name;
- (instancetype)initWithName:(NSString *)name;
- (NSString *)read:(NSString *)key;
@end

@implementation Store

- (instancetype)initWithName:(NSString *)name {
  self = [super init];
  if (self) {
    _name = [name copy];
  }
  return self;
}

- (NSString *)read:(NSString *)key {
  return [NSString stringWithFormat:@"%@:%@", self.name, key];
}

@end

static NSString *StoreDescription(Store *store) {
  return [store read:@"desc"];
}

int main(int argc, const char *argv[]) {
  @autoreleasepool {
    Store *store = [[Store alloc] initWithName:@"main"];
    NSLog(@"%@", StoreDescription(store));
  }
  return argc > 1 ? 1 : 0;
}

File: native/mixdog-graph/tests/fixtures/parity/swift/Sample.swift
// Parity fixture: declaration and import shapes the graph reports for Swift.
import Foundation
import class UIKit.UIView
@preconcurrency import Dispatch

public protocol Storage {
    func read(key: String) -> String?
}

public struct Point {
    public var x: Int
    public var y: Int

    public func sum() -> Int {
        return x + y
    }
}

public enum Mode {
    case fast
    case slow

    func label() -> String {
        return "\(self)"
    }
}

public class Store: Storage {
    let name: String

    public init(name: String) {
        self.name = name
    }

    public func read(key: String) -> String? {
        func nested(_ value: String) -> String {
            return value.trimmingCharacters(in: .whitespaces)
        }
        return nested("\(name):\(key)")
    }
}

public actor Coordinator {
    private var stores: [Store] = []

    func add(_ store: Store) {
        stores.append(store)
    }
}

func topLevel(view: UIView, queue: DispatchQueue) -> String {
    _ = (view, queue)
    return Date().description
}
